### ISSUE TITLE
Graphics problems on mac os x

### ISSUE THREAD

**Issue by ptiza-v-nebe** [colobot/colobot#778]
Hallo,
I have weird problem with shadows on an apple computer. Is it possible to solve this bug by just turning of some option in some cpp files? I have this problem on windows running through parallels too.
Or maybe just turning of this option... I dont know. Maybe it is just the driver? what maybe to do?
![screen shot 2016-05-13 at 23 02 59](https://cloud.githubusercontent.com/assets/5651265/15261900/4def31cc-195f-11e6-810b-1b515b1b92ab.png)

Thanks have a nice day!

**Comment by tomaszkax86** [comment]
Seems like low quality shadow map. Did you try changing shadow resolution?

**Comment by ptiza-v-nebe** [comment]
Yes I tryed, a little better but shadows still jitter very awful.

![screen shot 2016-05-14 at 08 36 13](https://cloud.githubusercontent.com/assets/5651265/15266765/bb18bea8-19af-11e6-81a5-0fde1469db41.png)


![screen shot 2016-05-14 at 08 36 08](https://cloud.githubusercontent.com/assets/5651265/15266760/a3b039bc-19af-11e6-88ca-7e18aa94c1bd.png)

![screen shot 2016-05-14 at 08 42 42](https://cloud.githubusercontent.com/assets/5651265/15266787/fc89911e-19af-11e6-9d9a-9a8130444a74.png)

**Comment by ptiza-v-nebe** [comment]
with normal Dynamic shadows I get blacks on sides that just suddently randomly popping up and disappear 

![screen shot 2016-05-14 at 08 49 09](https://cloud.githubusercontent.com/assets/5651265/15266822/1488c4b4-19b1-11e6-9583-9bca9dda782a.png)
![screen shot 2016-05-14 at 08 49 26](https://cloud.githubusercontent.com/assets/5651265/15266823/174d7e9c-19b1-11e6-8fb7-ecd097081e38.png)

**Comment by ptiza-v-nebe** [comment]
same with simple shadows
![screen shot 2016-05-14 at 08 52 17](https://cloud.githubusercontent.com/assets/5651265/15266832/3d1fed94-19b1-11e6-9f71-313fe07d6777.png)
![screen shot 2016-05-14 at 08 52 12](https://cloud.githubusercontent.com/assets/5651265/15266834/3ec4f81a-19b1-11e6-8d38-b65bc4ca7d08.png)

**Comment by ptiza-v-nebe** [comment]
Intel hd 3000 has opengl 3, I thought maybe opengl 4 would help. Tryed on another mac, with an intel hd 4000 and opengl 4 and have in mac os x still same thing

**Comment by ptiza-v-nebe** [comment]
tryed now on a macmini with bootcamp windows 8.1, intel hd 4000. simple shadows are ok, now random popping has disappeared. 
Dynamic Shadows makes same as on mac os x. Very very bad jittering of shadows on main character and on other objects. All with high shadow resolution.

**Comment by RaptorParkowsky** [comment]
It looks just normal to me. It's totally normal that you have pixelated shadow maps on those settings. Shadow mapping implementation is still pretty young and have some flaws. You can tweak it a little by changing one value in colobot.ini file.

I don't know where on OS X GOLD's save files are stored, so just search "colobot.ini" on your hard drive and open it with notepad. Find:
`ShadowRange=`

and change it's value for example to that:

`ShadowRange=128`

This simply decreases the standard shadowmap range, but shadows should look less pixelated.

**Comment by ptiza-v-nebe** [comment]
ok, maybe showing is better then 100 words

![jittering copy](https://cloud.githubusercontent.com/assets/5651265/15273144/9b71babe-1a90-11e6-8349-0cddbdfeb83a.gif)

I had never this problems on the lowest specs on a normal non apple pc

**Comment by ptiza-v-nebe** [comment]
ok, maybe showing is better then 100 words
![Uploading jittering3.gif…]()
I had never this problems on the lowest specs on a normal non apple pc

**Comment by RaptorParkowsky** [comment]
You'r photo didn't uploaded... :/

**Comment by ptiza-v-nebe** [comment]
now visible??

**Comment by krzys-h** [comment]
I was having always glitches like these since shadow mapping was implemented and I always assumed this is just a bug in the experimental implementation. Can anybody confirm this? @tomaszkax86 ?

**Comment by ptiza-v-nebe** [comment]
tryed cb on normal pc

simple shadows and dynamic shadows doing well no popping up of random shadows, no jittering. Besides that dynamic shadows are not antialiased on the ground.

Dynamic shadows ++ has the same effect as on a mac jittering without end 1:1.

simple shadow
![simpleshadows](https://cloud.githubusercontent.com/assets/5651265/15275832/83b219be-1ad6-11e6-863c-1ba21032ff41.PNG)

dynamic shadow
![dynamic shadows](https://cloud.githubusercontent.com/assets/5651265/15275881/d5c412ec-1ad7-11e6-8f9a-59127eb08faa.PNG)

dynamic shadow ++
![dynamic shadows](https://cloud.githubusercontent.com/assets/5651265/15275883/dcfc1172-1ad7-11e6-8e70-073dfb62d6c5.PNG)

comp specs
thinkpad i7, 8gb, intel hd graphics

**Comment by ptiza-v-nebe** [comment]
played with ShadowRange

ShadowRange=1024
![1024](https://cloud.githubusercontent.com/assets/5651265/15282430/9de9251c-1b43-11e6-9097-1ff37cf3a91c.gif)




ShadowRange=512
![512](https://cloud.githubusercontent.com/assets/5651265/15282433/a1718652-1b43-11e6-8a1c-c9a86494c715.gif)




ShadowRange=256
![256](https://cloud.githubusercontent.com/assets/5651265/15282438/a5595376-1b43-11e6-9353-d424ef2bc0d2.gif)




ShadowRange=128
![128](https://cloud.githubusercontent.com/assets/5651265/15282518/1a1cc1ac-1b44-11e6-8382-74bbd5b3caa0.gif)




ShadowRange=64
![64](https://cloud.githubusercontent.com/assets/5651265/15282561/70bf9aca-1b44-11e6-9b00-15aedd67584f.gif)




with more range its getting better and more playable

**Comment by dataf3l** [comment]
I compiled and installed the Software on this commit: 17447e813b97a68861c2ad257e7cead16724c16b, after running it on my mac with the instructions, I see some jitter (not a lot), but it seems to be ocluded by a battery pack.
I think the problem persists, but it is less noticeable than on these images. the problem is real.
